File: Pegasos.Web.Administrador/Services/AuthService.cs
using System.Net.Http.Headers;
using System.Text;
using System.Text.Json;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Microsoft.Extensions.Logging;
using System.Net.Http;
using System.Net.Http.Json;

namespace Pegasos.Web.Administrador.Services
{
    public class AuthService : IAuthService
    {
        private readonly HttpClient _httpClient;
        private readonly ILogger<AuthService> _logger;

        public AuthService(HttpClient httpClient, ILogger<AuthService> logger)
        {
            _httpClient = httpClient;
            _logger = logger;
        }


        public async Task<AuthResult?> LoginAsync(string username, string password)
        {
            try
            {
                _httpClient.DefaultRequestHeaders.Clear();

                var jsonManual = $"{{\"usuario\":\"{username}\",\"password\":\"{password}\"}}";
                var content = new StringContent(jsonManual, Encoding.UTF8, "application/json");

                _logger.LogInformation("Enviando login para usuario: {Username}", username);

                var response = await _httpClient.PostAsync("auth/login", content);

                if (response.IsSuccessStatusCode)
                {
                    var responseJson = await response.Content.ReadAsStringAsync();
                    _logger.LogInformation("Respuesta del login recibida: {Response}", responseJson);

                    var options = new JsonSerializerOptions
                    {
                        PropertyNameCaseInsensitive = true,
                        PropertyNamingPolicy = null // No cambiar nombres
                    };

                    var result = JsonSerializer.Deserialize<AuthResult>(responseJson, options);

                    if (result != null)
                    {
                        _logger.LogInformation("Token recibido: {TokenPresent} - Longitud: {Length}",
                            string.IsNullOrEmpty(result.access_token) ? "VACÍO" : "OK",
                            result.access_token?.Length ?? 0);

                        _logger.LogInformation("UsuarioID recibido: {UsuarioId}", result.usuarioID);
                    }

                    return result;
                }

                var error = await response.Content.ReadAsStringAsync();
                _logger.LogWarning("Login fallido - StatusCode: {StatusCode}, Error: {Error}",
                    response.StatusCode, error);
                return null;
            }
            catch (Exception ex)
            {
                _logger.LogError(ex, "Error de conexión en login");
                return null;
            }
        }

        public async Task<bool> ExtendSessionAsync(string accessToken)
        {
            try
            {
                _httpClient.DefaultRequestHeaders.Authorization =
                    new AuthenticationHeaderValue("Bearer", accessToken);

                var response = await _httpClient.PostAsync("auth/refresh", new StringContent("{}"));
                return response.IsSuccessStatusCode;
            }
            catch
            {
                return false;
            }
        }
    }
}

File: Pegasos.WEB.Admin/Services/AuthService.fs
namespace Pegasos.Web.Admin.Services

open System
open System.Net.Http
open System.Net.Http.Headers
open System.Text
open System.Text.Json
open System.Threading.Tasks
open Microsoft.Extensions.Logging

type AuthService(httpClient: HttpClient, logger: ILogger<AuthService>) =
    
    interface IAuthService with
        member this.LoginAsync(username, password) =
            async {
                try
                    let request = {| Username = username; Password = password |}
                    let json = JsonSerializer.Serialize(request)
                    let content = new StringContent(json, Encoding.UTF8, "application/json")

                    // Llamada al Gateway
                    let! response = httpClient.PostAsync("auth/login", content) |> Async.AwaitTask

                    if response.StatusCode = System.Net.HttpStatusCode.Unauthorized then
                        return None
                    
                    elif response.StatusCode = System.Net.HttpStatusCode.Created then
                        let! responseJson = response.Content.ReadAsStringAsync() |> Async.AwaitTask
                        let options = JsonSerializerOptions(PropertyNameCaseInsensitive = true)
                        let result = JsonSerializer.Deserialize<AuthResult>(responseJson, options)
                        return Some result
                    
                    else
                        logger.LogWarning("Login inesperado: {Status}", response.StatusCode)
                        return None
                with
                | ex ->
                    logger.LogError(ex, "Error en login")
                    return None
            } |> Async.StartAsTask

        member this.ExtendSessionAsync(accessToken) =
            async {
                try
                    httpClient.DefaultRequestHeaders.Authorization <- 
                        new AuthenticationHeaderValue("Bearer", accessToken)

                    let! response = httpClient.PostAsync("auth/refresh", new StringContent("{}")) |> Async.AwaitTask
                    return response.IsSuccessStatusCode
                with
                | _ -> return false
            } |> Async.StartAsTask